### ISSUE TITLE
some problem about softmax

### ISSUE THREAD

**Issue by 532091602hzy** [layumi/Person-reID-verification#10]
Thanks for sharing a wonderful code! This code don't have any issue and this is my private problem.
For positive pairs, the network final two-dimension FC layer output like [0.57, 0.43] vector (I add a softmax layer after original final FC layer). Even though the network can distinguish positive and negative (because positive pairs most greater than 0.5, negative opposite), for my own project it had better output positive close to 1 (such as 0.95).
So why this code final two-dimension FC layer output like [0.57, 0.43] rather than [0.95, 0.05] as usual softmax layer do.

**Comment by bmiftah** [comment]
I also thank the @layumi  for your time and energy to put this valuable resource.  I want to put quick question on why you replace the square layer with (x*y) direct product of feature vector , i.e why used  x*y instead of (x-y)^2 ?  what is the reasoning behind x*y - I run your code and it of course works with x*y .

**Comment by layumi** [comment]
Hi @bmiftah 
I found that `x^2` and `y^2` are hard to control when learning. Sometimes, they may lead to un-convergence. So I normalized x and y. In this way, minimizing '(x-y)^2' equals to maximizing the 'xy'.

**Comment by bmiftah** [comment]
Hi, Dear Zhedong Zheng, Thank you for your swift reply , I see , you use a substitute formulation to get the same thing without ignoring the variable x and y .i.e (x-y)^2=  x^2 - 2xy - y^2  =  (x^2 - y^2 - 2xy )  (ignoring 2 here and maximizing xy here  will minimize (x^2 - y^2 - 2xy ) ? ) 

so, in the model we still use same loss function after making this change , use cross-entropy - ?    It will achieve our objective for verification part ?
